hyp3_autorift/geometry.py

```diff
@@ -89,59 +86,37 @@ def bounding_box(safe, priority='reference', polarization='hh', orbits='Orbits',
     return lat_limits, lon_limits
 
 
-def find_jpl_dem(lat_limits, lon_limits, z_limits=(-200, 4000)):
+def polygon_from_bbox(lat_limits, lon_limits):
+    ring = ogr.Geometry(ogr.wkbLinearRing)
+    ring.AddPoint(lon_limits[0], lat_limits[0])
+    ring.AddPoint(lon_limits[1], lat_limits[0])
+    ring.AddPoint(lon_limits[1], lat_limits[1])
+    ring.AddPoint(lon_limits[0], lat_limits[1])
+    ring.AddPoint(lon_limits[0], lat_limits[0])
+    polygon = ogr.Geometry(ogr.wkbPolygon)
+    polygon.AddGeometry(ring)
+    return polygon
 
-    dems = ['GRE240m', 'ANT240m']
-    bounding_dem = None
-    for dem in dems:
-        dem_file = f'/vsicurl/http://{ITS_LIVE_BUCKET}.s3.amazonaws.com/{AUTORIFT_PREFIX}/{dem}_h.tif'
-        log.info(f'Checking DEM: {dem_file}')
-        dem_ds = gdal.Open(dem_file, gdal.GA_ReadOnly)
-        dem_sr = dem_ds.GetSpatialRef()
-        log.debug(f'DEM projection: {dem_sr}')
 
-        latlon = osr.SpatialReference()
-        latlon.ImportFromEPSG(4326)
+def find_jpl_dem(lat_limits, lon_limits):
+    shape_file = f'/vsicurl/http://{ITS_LIVE_BUCKET}.s3.amazonaws.com/{AUTORIFT_PREFIX}/autorift_parameters.shp'
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    shapes = driver.Open(shape_file, gdal.GA_ReadOnly)
 
-        trans = osr.CoordinateTransformation(latlon, dem_sr)
+    extent = polygon_from_bbox(lat_limits, lon_limits)
+    for feature in shapes.GetLayer(0):
+        if feature.geometry().Contains(extent.Centroid()):
```

Comment:
compute the centroid once prior to the loop

Reply:
good point! (see what I did there?)